Archive/Imo/Imo2025Q3.lean

```diff
@@ -0,0 +1,204 @@
+/-
+Copyright (c) 2025 Yi Yuan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yi Yuan
+-/
+import Mathlib.NumberTheory.LSeries.PrimesInAP
+import Mathlib.NumberTheory.LucasLehmer
+import Mathlib.NumberTheory.Multiplicity
+import Mathlib.Tactic.Simproc.Factors
+
+/-!
+# IMO 2025 Q3
+
+Let $\mathbb{N}+$ denote the set of positive integers. A function $f: \mathbb{N}+ \rightarrow
+\mathbb{N}+$ is said to be bonza if $f(a)$ divides $b^a-f(b)^{f(a)}$ for all positive integers $a$
+and $b$. Determine the smallest real constant $c$ such that $f(n) \leq c n$ for all bonza functions
+$f$ and all positive integers $n$.
+-/
```

Comment:
Please include an informal outline of the solution you're following in the module doc string (see the README.md for Archive/Imo).